packages/flutter/lib/src/material/icon.dart

```diff
@@ -65,13 +83,13 @@ class Icon extends StatelessComponent {
       category = parts[0];
       subtype = parts[1];
     }
-    // TODO(eseidel): This clearly isn't correct.  Not sure what would be.
-    // Should we use the ios images on ios?
-    String density = 'drawable-xxhdpi';
+    // TODO(kgiesing): Should selector logic be moved down a level so that
+    // any generic asset can be resolved using consistent contextual logic?
```

Comment:
I guess it's a question of how strong the policy is at the AssetBundle layer. The equivalent system on Android has the policy pretty deeply embedded, which is convenient, but sometimes not flexible.  Your suggestion means that developers have to create their own policy around DPI breakpoints.

An alternative would be to create a strong policy at the material layer which is more general than just Icon (a fairly limited use case even in material apps).